db/dbtypes/types.go

```diff
@@ -498,6 +498,103 @@ func ChoiceIndexFromStr(choice string) (VoteChoice, error) {
 	}
 }
 
+// ChartType defines the respective charts that appear on the /charts page. It also
+// helps define the respective indexes to the charts pushed into the cache array.
+type ChartType int8
+
+const (
+	// Pg Charts in alphabetical order.
+
+	AvgBlockSize ChartType = iota
+	BlockChainSize
+	ChainWork
+	CoinSupply
+	DurationBTW
+	HashRate
+	POWDifficulty
+	TicketByWindows
+	TicketPrice
+	TicketsByBlocks
+	TicketSpendT
+	TxPerBlock
+	TxPerDay
+
+	// Sqlite Charts in alphabetical order.
+
+	FeePerBlock
+	TicketPoolSize
+	TicketPoolValue
+)
+
+const (
+	// SqliteChartsCount defines the count of charts that use the sqlite db as
+	// data source and their data is held in cacheChartsData cache. When new
+	// sqlite charts are added, this value should be updated to avoid errors.
+	SqliteChartsCount = 3
+
+	// PgChartsCount defines the count of charts that use the auxiliary db as
+	// data source and their data is held in cacheChartsData cache. When new
+	// pg charts are added, this value should be updated to avoid errors.
+	PgChartsCount = 13
+)
+
+// ChartsDef maps the ChartType value to its respective chart name string.
+var ChartsDef = map[ChartType]string{
+	AvgBlockSize:    "avg-block-size",
+	BlockChainSize:  "blockchain-size",
+	ChainWork:       "chainwork",
+	CoinSupply:      "coin-supply",
+	DurationBTW:     "duration-btw-blocks",
+	FeePerBlock:     "fee-per-block",
+	HashRate:        "hashrate",
+	POWDifficulty:   "pow-difficulty",
+	TicketByWindows: "ticket-by-outputs-windows",
+	TicketPoolSize:  "ticket-pool-size",
+	TicketPoolValue: "ticket-pool-value",
+	TicketPrice:     "ticket-price",
+	TicketsByBlocks: "ticket-by-outputs-blocks",
+	TicketSpendT:    "ticket-spend-type",
+	TxPerBlock:      "tx-per-block",
+	TxPerDay:        "tx-per-day",
+}
```

Comment:
More charts will added after this PR but you raise a valid point.

Reply:
>I'd much rather have `ChartType` be `string` itself, and the data cache be a `map[ChartType]*ChartsData` similar to the way it was implemented before.

Using a `string` to reference charts IMO introduces a possible bug in case one references a non-existent chart while assigning or fetching data.

Reply:
I agree with @buck54321 about the map.

Reply:
Okay. I will revert back to the map implementation.